sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -917,30 +971,6 @@ public void startBundle() {
         putFutures = Collections.synchronizedList(new ArrayList<>());
         /** Keep only the first {@link MAX_NUM_FAILURES} occurred exceptions */
         failures = new LinkedBlockingDeque<>(MAX_NUM_FAILURES);
-        initKinesisProducer();
-      }
-
-      private synchronized void initKinesisProducer() {
```

Comment:
Yet another issue is synchronisation when creating the producer instance. If the producer is statically shared among writers of the same JVM, synchronisation cannot be done on `this`.
Otherwise there might be multiple producer deamons running, with only the latest one being used.